### PR TITLE
Embedding code snippets

### PR DESCRIPTION
**Pull request by jtulach**
Once https://github.com/graalvm/mx/pull/65 is accepted, we can start using it. This is the rewrite of `{@codesnippet ...}` tags to `{@link ...Snippets...}` tags that can be easily navigated from the IDE. This will address problems reported in https://github.com/graalvm/truffle/issues/19

### CODE REVIEW

**Comment by jtulach** [comment]
Ready for integration.

File: truffle/com.oracle.truffle.api.profiles/src/com/oracle/truffle/api/profiles/BranchProfile.java [hunk 1]
```diff
@@ -131,3 +132,22 @@ public String toString() {
     }
 
 }
+
+class BranchProfileSnippets {
+    class Node {
+    }
```
Comment (chumer) [inline truffle/com.oracle.truffle.api.profiles/src/com/oracle/truffle/api/profiles/BranchProfile.java]: Why don't you use the real Node?

**Comment by jtulach** [comment]
True, in this case we can use real Node. Probably copy paste error. OK now?

File: truffle/com.oracle.truffle.api.vm/src/com/oracle/truffle/api/vm/PolyglotEngine.java [hunk 2]
```diff
@@ -1285,3 +1285,21 @@ protected String toString(TruffleLanguage language, TruffleLanguage.Env env, Obj
...
+        // BEGIN: com.oracle.truffle.api.vm.PolyglotEngineSnippets#initializeWithParameters
+        String[] args = {"--kernel", "Kernel.som", "--instrument", "dyn-metrics"};
+        PolyglotEngine.Builder builder = PolyglotEngine.newBuilder();
+        builder.config(YourLang.MIME_TYPE, "CMD_ARGS", args);
+        PolyglotEngine vm = builder.build();
+        // END: com.oracle.truffle.api.vm.PolyglotEngineSnippets#initializeWithParameters
+        // @formatter:on
+        return vm;
+    }
+}
```
Comment (woess) [inline truffle/com.oracle.truffle.api.vm/src/com/oracle/truffle/api/vm/PolyglotEngine.java]: No newline at the end of the file. How come checkstyle does not complain about it? Is it not enabled for the truffle.api.vm project?

Comment (woess) [inline truffle/com.oracle.truffle.api.vm/src/com/oracle/truffle/api/vm/PolyglotEngine.java]: It really isn't checked by checkstyle. Please enable it (add the following line in suite.py).
`"checkstyle" : "com.oracle.truffle.api",`

Comment (jtulach) [inline truffle/com.oracle.truffle.api.vm/src/com/oracle/truffle/api/vm/PolyglotEngine.java]: Good catch.

**Comment by chumer** [comment]
Its a shame that eclipse cannot properly handle doclets. Having the snippets that close definitely makes it simpler to find the examples when browsing code. While this use-case still is not perfect (eg. clicking on the link will show you the javadoc of the snippet and not the code), I think its good enough now to be used. The formatted code in javadoc looks very good.

I've also verified that the snippets code does not appear in the distribution jars.

**Pull request merged**